pkg/server/application_api/application_api_test: TestClusterResetSQLStats failed

pkg/server/application_api/application_api_test.TestClusterResetSQLStats [failed](https://mesolite.cluster.engflow.com/invocations/default/b46d62ef-e718-45b7-a0b6-b7d2fe4d9e6b?testReportRun=2&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyL2FwcGxpY2F0aW9uX2FwaTphcHBsaWNhdGlvbl9hcGlfdGVzdA==) on release-24.3 @ [52d6d7fb924c0e18525e3bb420f06d2eb1b96fa7](https://github.com/cockroachdb/cockroach/commits/52d6d7fb924c0e18525e3bb420f06d2eb1b96fa7):


```
=== RUN   TestClusterResetSQLStats
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestClusterResetSQLStats1578952351
    test_log_scope.go:76: use -show-logs to present logs inline
    stats_test.go:387: -- test log scope end --
test logs left over in: outputs.zip/logTestClusterResetSQLStats1578952351
--- FAIL: TestClusterResetSQLStats (82.85s)
=== RUN   TestClusterResetSQLStats/flushed=false
    test_server_shim.go:152: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    stats_test.go:367: expected to find stats last reset value changed, but didn't
    --- FAIL: TestClusterResetSQLStats/flushed=false (36.06s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestClusterResetSQLStats.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-64181

--- a/pkg/server/application_api/sql_stats_test.go
+++ b/pkg/server/application_api/sql_stats_test.go
@@ -1805,6 +1805,7 @@ func TestClusterResetSQLStats(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	skip.UnderDuress(t)
+	skip.UnderRace(t, "flaky under race due to RPC fanout timing; see #170803")
 
 	ctx := context.Background()
 	knobs := sqlstats.CreateTestingKnobs()

